### ISSUE TITLE
IDE0028/IDE0300 is not reported for overloaded methods

### ISSUE THREAD

**Issue by cremor** [dotnet/roslyn#73362]
**Version Used**: Visual Studio 17.9.6, SDK 8.0.204

**Steps to Reproduce**:
.editorconfig:
```
dotnet_style_prefer_collection_expression = when_types_loosely_match
```

Code:
```cs
    public void Test(Class1 param1, Class1 param2)
    {
        // Should show IDE0300 but doesnt't
        MethodTakingEnumerable(new[] { param1, param2 });
        MethodTakingEnumerable(new Class1[] { param1, param2 });

        // Should show IDE0028 but doesnt't
        MethodTakingEnumerable(new List<Class1> { param1, param2 });
    }

    public void MethodTakingEnumerable(IEnumerable<Class1> param)
    {
    }

    public void MethodTakingEnumerable(IEnumerable<Class2> param)
    {
    }

    public class Class1 { }
    public class Class2 { }
```

**Diagnostic Id**:
`IDE0028`: Use collection initializers or expressions
`IDE0300`: Use collection expression for array

**Expected Behavior**:
Method calls to overloaded methods should show IDE0028/IDE0300 analyzer diagnostics.

**Actual Behavior**:
Method calls to overloaded methods do not show IDE0028/IDE0300 analyzer diagnostics.

**Comment by cremor** [comment]
Should this be tagged with the label "New Feature - Collection Expressions"?

**Comment by CyrusNajmabadi** [comment]
No repro:

![Image](https://github.com/user-attachments/assets/26bdb062-a045-4b21-89ff-b856e0d4f3ec)

![Image](https://github.com/user-attachments/assets/d1cf4c1f-8f85-4ef3-a2cd-d799e7567a40)

**Comment by cremor** [comment]
> No repro:

@CyrusNajmabadi I can still reproduce all 3 cases in VS 17.11.4 with SDK 8.0.400
